lib/repl.js

```diff
@@ -928,6 +951,16 @@ function complete(line, callback) {
   function completionGroupsLoaded(err) {
     if (err) throw err;
 
+    if (warning.type) {
+      self._writeToOutput('\r\n');
+      process.emitWarning(
+        warning.message,
+        warning.type,
+        undefined,
+        undefined,
+        true);
```

Comment:
refer https://github.com/nodejs/node/blob/v8.1.2/lib/internal/process/warning.js#L153

if it's true, it won't be emitted on next tick.